modules/setting/attachment.go

```diff
@@ -32,37 +25,9 @@ var (
 
 func newAttachmentService() {
 	sec := Cfg.Section("attachment")
-	Attachment.Storage.Type = sec.Key("STORAGE_TYPE").MustString("")
-	if Attachment.Storage.Type == "" {
-		Attachment.Storage.Type = "default"
-	}
+	storageType := sec.Key("STORAGE_TYPE").MustString("")
 
-	if Attachment.Storage.Type != LocalStorageType && Attachment.Storage.Type != MinioStorageType {
-		storage, ok := storages[Attachment.Storage.Type]
-		if !ok {
-			log.Fatal("Failed to get attachment storage type: %s", Attachment.Storage.Type)
-		}
-		Attachment.Storage = storage
-	}
-
-	// Override
-	Attachment.ServeDirect = sec.Key("SERVE_DIRECT").MustBool(Attachment.ServeDirect)
-
-	switch Attachment.Storage.Type {
-	case LocalStorageType:
-		Attachment.Path = sec.Key("PATH").MustString(filepath.Join(AppDataPath, "attachments"))
-		if !filepath.IsAbs(Attachment.Path) {
-			Attachment.Path = filepath.Join(AppWorkPath, Attachment.Path)
-		}
-	case MinioStorageType:
-		Attachment.Minio.Endpoint = sec.Key("MINIO_ENDPOINT").MustString(Attachment.Minio.Endpoint)
-		Attachment.Minio.AccessKeyID = sec.Key("MINIO_ACCESS_KEY_ID").MustString(Attachment.Minio.AccessKeyID)
-		Attachment.Minio.SecretAccessKey = sec.Key("MINIO_SECRET_ACCESS_KEY").MustString(Attachment.Minio.SecretAccessKey)
-		Attachment.Minio.Bucket = sec.Key("MINIO_BUCKET").MustString(Attachment.Minio.Bucket)
-		Attachment.Minio.Location = sec.Key("MINIO_LOCATION").MustString(Attachment.Minio.Location)
-		Attachment.Minio.UseSSL = sec.Key("MINIO_USE_SSL").MustBool(Attachment.Minio.UseSSL)
-		Attachment.Minio.BasePath = sec.Key("MINIO_BASE_PATH").MustString("attachments/")
-	}
+	Attachment.Storage = getStorage("attachment", storageType, sec)
```

Comment:
How did we keep these compitable?
```
Attachment.Path = sec.Key("PATH").MustString(filepath.Join(AppDataPath, "attachments"))
		if !filepath.IsAbs(Attachment.Path) {
			Attachment.Path = filepath.Join(AppWorkPath, Attachment.Path)
		}
```

Reply:
Sorry this should be

```suggestion
	Attachment.Storage = getStorage("attachments", storageType, sec)
```